kafka-rest/src/main/java/io/confluent/kafkarest/v2/KafkaConsumerManager.java

```diff
@@ -390,7 +387,10 @@ public RunnableReadTask(ReadTaskState taskState, KafkaRestConfig config) {
      * Delays for a minimum of {@code delayMs} or until the read request expires
      */
     public void delayFor(Duration delay) {
-      if (!requestExpiration.isAfter(clock.instant())) {
+      // the first condition means that the request has expired,
+      // the second one means that the request has been retried for
+      // the last time, so if it goes to this block again, it will be expired
+      if (!requestExpiration.isAfter(clock.instant()) || requestExpiration.equals(waitExpiration)) {
```

Comment:
this is the main fix for the flakiness

Reply:
Why does this fix the problem?

Reply:
@ehumber I've provided the detailed explanation in the description 👍 , hope it is clear enough.